src/SqlBindingUtilities.cs

```diff
@@ -290,5 +292,65 @@ internal static bool IsFatalSqlException(this Exception e)
             }
             return true;
         }
+
+        /// <summary>
+        /// Get the Server Properties for the given connection.
+        /// </summary>
+        /// <returns>ServerProperties (EngineEdition and Edition) of the target Sql Server.</returns>
+        public static async Task<ServerProperties> GetSqlServerEditions(SqlConnection connection, ILogger logger, CancellationToken cancellationToken)
+        {
+            if (TelemetryInstance.Enabled)
+            {
+                string selectServerEdition = $"SELECT SERVERPROPERTY('EngineEdition'), SERVERPROPERTY('EngineEdition')";
```

Comment:
Put all this in a try/catch - if an exception happens log it then just return null